r/R/dplyr-funcs.R

```diff
@@ -58,15 +58,34 @@ NULL
 #' @keywords internal
 #'
 register_binding <- function(fun_name, fun, registry = nse_funcs) {
-  name <- gsub("^.*?::", "", fun_name)
-  namespace <- gsub("::.*$", "", fun_name)
+  qualified_name <- fun_name
```

Comment:
I think you can simplify this function if you just deal with `::` separately. (I'll suggest below.) There's a lot of special casing that you could drop.